### PR TITLE
Allows to add credentials user export

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

The user import file can now be manually modified to add credentials for user.

Also the `--generate-credentials` option allows to automatically generate credentials for exported users.

```
USAGE
  $ kourou user:export

OPTIONS
  --as=as                                  Impersonate a user
  --batch-size=batch-size                  [default: 2000] Maximum batch size (see limits.documentsFetchCount config)
  --exclude=exclude                        Exclude users by matching their IDs with a regexp
  --generate-credentials                   Generate credentials with a random password for users
  --generated-username=generated-username  [default: _id] User content property used as a username for local credentials
  --help                                   show CLI help
  --host=host                              [default: localhost] Kuzzle server host
  --password=password                      Kuzzle user password
  --path=path                              [default: users] Dump directory
  --port=port                              [default: 7512] Kuzzle server port
  --protocol=protocol                      [default: ws] Kuzzle protocol (http or websocket)
  --ssl                                    Use SSL to connect to Kuzzle
  --username=username                      [default: anonymous] Kuzzle username (local strategy)

DESCRIPTION
  Exports users to JSON.

  The users will be exported WITHOUT their credentials since Kuzzzle can't access them.

  You can either:
     - manually re-create credentials for your users
     - use the "mustChangePasswordIfSetByAdmin" option Kuzzle password policies (see 
  https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/#optional-properties)
     - use the "--generate-credentials" flag to auto-generate credentials for your users

  Auto-generation of credentials

     With the "--generate-credentials" flag, Kourou will adds credentials for the "local" strategy.
     By default, the username will be the user ID.
     Use the "generated-username" flag to use an other property than the user ID for the generated username
     The password will be a strong random 40 characters string

  Examples:

     - kourou user:export
     - kourou user:export --exclude '.*admin.*' --exclude 'supervisor.*'
     - kourou user:export --generate-credentials
     - kourou user:export --generate-credentials --generated-username content.email
```

### How should this be manually tested?

Try the provided examples :arrow_up:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -958,24 +958,51 @@ _See code: [src/commands/subscribe.ts](https://github.com/kuzzleio/kourou/blob/v
...
+
+  You can either:
+     - manually re-create credentials for your users
+     - use the "mustChangePasswordIfSetByAdmin" option Kuzzle password policies (see 
+  https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/#optional-properties)
+     - use the "--generate-credentials" flag to auto-generate credentials for your users
+
+  Auto-generation of credentials
+
+     With the "--generate-credentials" flag, Kourou will adds credentials for the "local" strategy.
```
Comment (xbill82) [inline README.md:995]:
```suggestion
     With the "--generate-credentials" flag, Kourou will add credentials for the "local" strategy.
```

File: src/commands/user/export.ts [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
+The users will be exported WITHOUT their credentials since Kuzzzle can't access them.
+
+You can either:
+  - manually re-create credentials for your users
+  - use the "mustChangePasswordIfSetByAdmin" option Kuzzle password policies (see https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/#optional-properties)
+  - use the "--generate-credentials" flag to auto-generate credentials for your users
+
+Auto-generation of credentials
+
+  With the "--generate-credentials" flag, Kourou will adds credentials for the "local" strategy.
```
Comment (xbill82) [inline src/commands/user/export.ts:27]:
```suggestion
  With the "--generate-credentials" flag, Kourou will add credentials for the "local" strategy.
```

File: src/support/restore-securities.ts [hunk 3]
```diff
@@ -1,4 +1,9 @@
 import * as _ from 'lodash'
+import * as Bluebird from 'bluebird'
+
+// I can't find correct Bluebird types, I tried the package but he doesn't knows
+// Bluebird.map
+const OiseauBleu: any = Bluebird
```
Comment (xbill82) [inline src/support/restore-securities.ts:6]: WAT
The `@types/bluebird` doesn't work?

Comment (xbill82) [reply src/support/restore-securities.ts:6]: BTW we should try to code an entire application with full french naming.

File: src/commands/user/export.ts [hunk 4]
```diff
@@ -1,13 +1,41 @@
...
+
 export default class UserExport extends Kommand {
-  static description = 'Exports users'
+  static description = `
+Exports users to JSON.
+
+The users will be exported WITHOUT their credentials since Kuzzzle can't access them.
+
+You can either:
+  - manually re-create credentials for your users
```
Comment (Leodau) [inline src/commands/user/export.ts:21]:
```suggestion
  - Manually re-create credentials for your users
```

File: src/commands/user/export.ts [hunk 5]
```diff
@@ -1,13 +1,41 @@
...
+  - use the "mustChangePasswordIfSetByAdmin" option Kuzzle password policies (see https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/#optional-properties)
```
Comment (Leodau) [inline src/commands/user/export.ts:22]:
```suggestion
  - Use the "mustChangePasswordIfSetByAdmin" option Kuzzle password policies (see https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/#optional-properties)
```

File: src/commands/user/export.ts [hunk 6]
```diff
@@ -1,13 +1,41 @@
...
+  - use the "--generate-credentials" flag to auto-generate credentials for your users
```
Comment (Leodau) [inline src/commands/user/export.ts:23]:
```suggestion
  - Use the "--generate-credentials" flag to auto-generate credentials for your users
```

File: src/support/restore-securities.ts [hunk 7]
```diff
@@ -1,4 +1,9 @@
 import * as _ from 'lodash'
+import * as Bluebird from 'bluebird'
+
+// I can't find correct Bluebird types, I tried the package but he doesn't knows
```
Comment (Leodau) [inline src/support/restore-securities.ts:4]:
```suggestion
```

File: src/support/restore-securities.ts (hunk 3 above)
Comment (rolljee) [reply src/support/restore-securities.ts:6]: I've just tried it on sandbox it worked for me as expected.
If you remove `:any` type and hover the OiseauBlue variable does it say  

image

Comment (Aschen) [reply src/support/restore-securities.ts:6]: @rolljee Have you tried to call the `Bluebird.map` method?

**Pull request merged**